### DISCUSSION TITLE
Restart Libreoffice before the next new request

### DISCUSSION THREAD

**Discussion by bahag-xvisnei** [gotenberg/gotenberg#782] [Q&A] [answered]
Hi,
--libreoffice-restart-after controls after how many requests Libreoffice process is going to be restarted. A restart happens at the next request after the configured --libreoffice-restart-after ocurrences. A restart takes multiple seconds. Would it be not better to restart right after the last request so when the next request arrives, it could be processed faster without waiting.

Bests

**Comment by gulien** [comment]
Hello @bahag-xvisnei,

Under constant load, the solution your proposing does not change much, as I've implemented a lock mechanism to prevent concurrent conversion. It would only change which request is "waiting" (i.e., in your proposal, the next request in queue).

**Reply by bahag-xvisnei** [reply]
Hi @gulien ,

I agree. Under constant load, it won't make much difference. In my case, though, it could make a difference, because a pdf is generated upon a user click. As I don't expect multiple concurrent requests, a user could benefit from this small change.

Overall, it has not a huge impact on my app. I just wanted to give a feedback.

Bests

**Reply by gulien** [reply, marked as answer]
I could work on something, but currently I have other stuff on my plate 😄
